### PR TITLE
Add validate function for token target qty v2

### PR DESCRIPTION
**Pull request by halink0803**

### CODE REVIEW

**Comment by halink0803** [comment]
[Validate TargetQuantity V2 input](https://trello.com/c/Ni7I67ob/134-validate-targetquantity-v2-input)

File: data/storage/bolt.go [hunk 1]
```diff
@@ -1423,10 +1423,11 @@ func (self *BoltStorage) RemovePendingStableTokenParams() error {
 	return err
 }
 
+//StorePendingTargetQtyV2 store value into pending target qty v2 bucket
 func (self *BoltStorage) StorePendingTargetQtyV2(value []byte) error {
 	var (
 		err  error
-		temp = make(map[string]interface{})
+		temp metric.TokenTargetQtyV2
```
Comment (favadi) [inline data/storage/bolt.go]: should we named it something else as we have defined a struct for it?

File: data/storage/bolt.go [hunk 2]
```diff
@@ -1458,7 +1460,7 @@ func (self *BoltStorage) GetPendingTargetQtyV2() (map[string]interface{}, error)
 }
 
 func (self *BoltStorage) ConfirmTargetQtyV2(value []byte) error {
-	temp := make(map[string]interface{})
+	temp := metric.TokenTargetQtyV2{}
```
Comment (favadi) [inline data/storage/bolt.go]: should we named it something else as we have defined a struct for it?

File: metric/types.go [hunk 3]
```diff
@@ -63,7 +66,30 @@ type TargetQtySet struct {
 
 //TargetQtyStruct object for save target qty
 type TargetQtyStruct struct {
```
Comment (favadi) [inline metric/types.go]: TargetQtyStruct is not a nice name as it is a struct already.

File: metric/types.go [hunk 4]
```diff
@@ -63,7 +66,30 @@ type TargetQtySet struct {
...
+	ExchangeRatio    map[string]float64 `json:"exchange_ratio,omitempty"`
+}
+
+//TokenTargetQtyV2 represent a map of token and its target quantity struct
+type TokenTargetQtyV2 map[string]TargetQtyStruct
+
+//IsValid validate token target quantity input
+func (tq TokenTargetQtyV2) IsValid() (bool, error) {
+	for k := range tq {
+		log.Print(k)
```
Comment (favadi) [inline metric/types.go]: please remove this line of log or make it nicer.

File: metric/types.go [hunk 5]
```diff
@@ -63,7 +66,30 @@ type TargetQtySet struct {
...
+		token, err := common.GetInternalToken(k)
+		log.Printf("Token: %v, err: %v", token, err)
+		if _, err := common.GetInternalToken(k); err != nil {
```
Comment (favadi) [inline metric/types.go]: `common.GetInternalToken` call are duplicated, maybe remove one?

File: metric/types.go [hunk 6]
```diff
@@ -63,7 +66,30 @@ type TargetQtySet struct {
...
+			return false, fmt.Errorf("Token %s is not supported", k)
+		}
+	}
+	return true, nil
+}
+
+//IsValid function validate target quantity struct
+func (tq TargetQtyStruct) IsValid() error {
+	return nil
```
Comment (favadi) [inline metric/types.go]: not sure if this function is useful.

**Pull request merged**